File: Examples/HelloWorld-iOS/HelloWorld-iOS/ViewController.m
//
//  ViewController.m
//  HelloWorld-iOS
//
//  Created by Cătălin Stan on 11/8/15.
//  Copyright © 2015 Catalin Stan. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UITextView *logTextView;

- (NSDictionary*)linkTextAttributes;

@end

@implementation ViewController

- (NSDictionary*)linkTextAttributes {
    static NSDictionary* _linkTextAttributes;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        _linkTextAttributes = @{
                                NSForegroundColorAttributeName: [UIColor blueColor],
                                NSUnderlineStyleAttributeName: @(NSUnderlineStyleSingle),
                                };
    });
    return _linkTextAttributes;
}

- (void)viewDidLoad {
    [super viewDidLoad];

    // Eye-candy
    self.logTextView.text = nil;
    self.logTextView.attributedText = nil;
    self.logTextView.linkTextAttributes = self.linkTextAttributes;

    [self setNeedsStatusBarAppearanceUpdate];

    // Append message to the list
    [[NSNotificationCenter defaultCenter] addObserverForName:@"LogMessage" object:nil queue:nil usingBlock:^(NSNotification * _Nonnull note) {
        NSAttributedString* attributedString = note.object;
        [self.logTextView.textStorage appendAttributedString:attributedString];
        [self.logTextView scrollRangeToVisible:NSMakeRange(self.logTextView.text.length, 0)];
    }];

}

- (BOOL)prefersStatusBarHidden {
    return YES;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Criollo/Apps/CriolloiOSApp/ViewController.swift
//
//  ViewController.swift
//  CriolloiOSApp
//
//  Created by Cătălin Stan on 27/03/16.
//  Copyright © 2016 Cătălin Stan. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITextViewDelegate {

    @IBOutlet weak var logTextView: UITextView!;

    @IBOutlet weak var statusImageItem: UIBarButtonItem!;
    @IBOutlet weak var startItem: UIBarButtonItem!;
    @IBOutlet weak var stopItem: UIBarButtonItem!;

    @IBOutlet weak var toolbar: UIToolbar!;
    @IBOutlet weak var statusDetailsButton: UILabel!;

    weak var appDelegate: AppDelegate!;
    strong var linkChecker: NSDataDetector!;

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

